### ISSUE TITLE
Vivecraft compatibility

### ISSUE THREAD

**Issue by Dunk0w0** [qouteall/ImmersivePortalsMod#1427]
### Is it a mod compatibility issue?

Yes

### If it's a mod compatibility issue, please provide the link(s) to the incompatible mod(s)

https://modrinth.com/mod/vivecraft

### Minecraft version

1.20.1

### Describe the issue

Going through portals in VR using Vivecraft is extremely buggy. Traveling vertically with portals works, but has some visual bugs. But when traveling horizontally, especially with longer distances, it both visually looks strange but also frequently misplaces the player. Similar issue occurs when traveling across dimensions.

[latest - Copy.log](https://github.com/iPortalTeam/ImmersivePortalsMod/files/12423277/latest.-.Copy.log)

https://github.com/iPortalTeam/ImmersivePortalsMod/assets/114096969/19ba6aff-39b3-469f-9aae-951f1711ad72

https://github.com/iPortalTeam/ImmersivePortalsMod/assets/114096969/4ac99d0e-cd57-4fe4-aa41-7a66038cbf93

https://github.com/iPortalTeam/ImmersivePortalsMod/assets/114096969/0ad7a51e-8966-45ca-b0e2-29d015288eb8

It would be really fun to see this fixed!! :)

**Comment by qouteall** [comment]
Someone reported that in old versions, the portal doesn't even render with vivecraft. Seem to be a progress?

**Comment by Dunk0w0** [comment]
Yeah the portals render fine on the fabric version, it's on forge that the rendering completely breaks.

**Comment by Nathan22211** [comment]
I'm having this too, they're completely black with complementary reimagined + vivecraft. but what's odd is that when I go through there's no portal on the other side. I tried with the vanilla nether portal on 1.20.1

**Comment by Merith-TK** [comment]
would like to update this with having an "simular" issue on Fabric
Portal has no issues rendering, doesnt break with travel

except, entering an portal will teleport you to your overworld chords in the desired dimension

so entering an nether portal at 1000X, and 1000Z will bring you to 1000,1000 in the nether, except the portal will be where it was supposed to be

This only happens when the user is in VR

**Comment by brzewVCE** [comment]
I would like to add my two cents. 

> I'm having this too, they're completely black with complementary reimagined + vivecraft. but what's odd is that when I go through there's no portal on the other side. I tried with the vanilla nether portal on 1.20.1

Works for my 1.20.2 when you enable Compatibility Rendering in config

> would like to update this with having an "simular" issue on Fabric Portal has no issues rendering, doesnt break with travel
> 
> except, entering an portal will teleport you to your overworld chords in the desired dimension
> 
> so entering an nether portal at 1000X, and 1000Z will bring you to 1000,1000 in the nether, except the portal will be where it was supposed to be
> 
> This only happens when the user is in VR

Also happens to me

**Comment by dasanibroto** [comment]
In 1.20.4, a lot of incompatible things work for me. I'm using vivecraft AND slidurs vibrant shaders, both of which are apparently incompatible. However (with compatibility rendering enabled), everything seems to work fine. 

https://github.com/iPortalTeam/ImmersivePortalsMod/assets/6785746/225de65d-4953-4506-9281-1fb1d4828d7d

**Comment by dasanibroto** [comment]
Never mind about the shader pack report, it still bugs the shadows when using custom portals.

**Comment by Nanorasmus** [comment]
I would like to add to this and say I personally don't get the rendering bugs when LOOKING through a portal (Though I do play with compatibility rendering turned on), however EVERY SINGLE TIME I walk through a portal one of 3 things happen:
1. I go through, but rendering bugs and all non-ui elements are complete darkness, this is fixed if I somehow walk back through the portal while blind (The portal is also pitch darkness, can't see it, but just walking backwards through it fixes it).
2. I get teleported ~10 blocks above the original portal I entered, fall down, and take fall damage, but still at approximately the same location I started.
3. I get teleported to whatever dimension the portal led to correctly, but at a seemingly random spot, on overworld2overworld portal this is often within a 100 blocks of the correct destination (Also often inside a wall, although suffocation damage doesn't seem to trigger)

For debugging purposes I'm playing on Quilt version 0.26.0 (MC 1.19.2) with a decent bit of mods, but all that are relevant to this issue are:
- Vivecraft 1.1.9
- Immersive Portals 2.3.2
- HexCasting 0.10.3 (A programming-esque magic mod)
- HexWays 6.0.1 (A HexCasting addon that allows the caster to spawn in immersive portal's portals)

**Comment by Nanorasmus** [comment]
Alrighty all, I made my own little mod to (somewhat) patch this issue, currently under review by Modrinth:
https://modrinth.com/mod/immersiveportals-vivecraft-compatibility

TO THE DEVS: It's not clean enough for a PR IMO but if you want the code you are free to steal it:
https://github.com/Nanorasmus/ImmesivePortalsVivecraftCompatibility

**Comment by Kat-09** [comment]
> Alrighty all, I made my own little mod to (somewhat) patch this issue, currently under review by Modrinth: https://modrinth.com/mod/immersiveportals-vivecraft-compatibility
> 
> TO THE DEVS: It's not clean enough for a PR IMO but if you want the code you are free to steal it: https://github.com/Nanorasmus/ImmesivePortalsVivecraftCompatibility

any plans to port this patch to 1.20.1?

**Comment by Nanorasmus** [comment]
> > Alrighty all, I made my own little mod to (somewhat) patch this issue, currently under review by Modrinth: https://modrinth.com/mod/immersiveportals-vivecraft-compatibility
> > TO THE DEVS: It's not clean enough for a PR IMO but if you want the code you are free to steal it: https://github.com/Nanorasmus/ImmesivePortalsVivecraftCompatibility
> 
> any plans to port this patch to 1.20.1?

I'll see what I can do! :D

**Comment by Anonimous22848** [comment]
> Someone reported that in old versions, the portal doesn't even render with vivecraft. Seem to be a progress?
On minecraft version 1.20.4 all was really good, everything was rendered as without vivecraft game (all about custom portals, i'm not try ender or nether portal). In addition, addons for I.P. (like pehkui and gravity changer mod) also works really well, but changing player size just stops you for a second in portal, and with changing gravity, gravity changes only for HDM view, hands will be in "normal" gravity (sorry for vr-skill issue, i downloaded vivecraft like hour ago.
(I cannot send video to this message because github is crap)
